openverse_catalog/dags/providers/provider_api_scripts/flickr.py

```diff
@@ -59,345 +36,258 @@
+            "filetype": filetype,
+            "creator": creator,
+            "creator_url": creator_url,
+            "title": title,
+            "meta_data": meta_data,
+            "raw_tags": raw_tags,
+            "source": source,
+            "category": category,
         }
-    )
-
-    return query_param_dict
-
-
-def _extract_image_list_from_json(response_json):
-    if response_json is None or response_json.get("stat") != "ok":
-        image_page = None
-    else:
-        image_page = response_json.get("photos")
-
-    if image_page is not None:
-        image_list = image_page.get("photo")
-        total_pages = image_page.get("pages")
-    else:
-        image_list, total_pages = None, None
-
-    return image_list, total_pages
-
-
-def _process_image_list(image_list):
-    total_images = 0
-    for image_data in image_list:
-        total_images = _process_image_data(image_data)
-
-    return total_images
-
-
-def _process_image_data(image_data, sub_providers=SUB_PROVIDERS, provider=PROVIDER):
-    logger.debug(f"Processing image data: {image_data}")
-    image_url, height, width = _get_image_url(image_data)
-    license_, license_version = _get_license(image_data.get("license"))
-    creator_url = _build_creator_url(image_data)
-    foreign_id = image_data.get("id")
-    if foreign_id is None:
-        logger.warning("No foreign_id in image_data!")
-    foreign_landing_url = _build_foreign_landing_url(creator_url, foreign_id)
-    owner = image_data.get("owner").strip()
-    source = next((s for s in sub_providers if owner in sub_providers[s]), provider)
-    filesize, filetype = _get_file_properties(image_url)
-    return image_store.add_item(
-        foreign_landing_url=foreign_landing_url,
-        image_url=image_url,
-        license_info=get_license_info(
-            license_=license_, license_version=license_version
-        ),
-        foreign_identifier=foreign_id,
-        width=width,
-        height=height,
-        filesize=filesize,
-        filetype=filetype,
-        creator=image_data.get("ownername"),
-        creator_url=creator_url,
-        title=image_data.get("title"),
-        meta_data=_create_meta_data_dict(image_data),
-        raw_tags=_create_tags_list(image_data),
-        source=source,
-        category=_get_category(image_data),
-    )
-
-
-def _build_creator_url(image_data, photo_url_base=PHOTO_URL_BASE):
-    owner = image_data.get("owner")
-    if owner is not None:
-        creator_url = _url_join(photo_url_base, owner.strip())
-        logger.debug(f"creator_url: {creator_url}")
-    else:
-        logger.warning("No creator_url constructed!")
-        creator_url = None
-
-    return creator_url
 
+    def _url_join(self, *args):
+        return "/".join([s.strip("/") for s in args])
+
+    @staticmethod
+    def _get_largest_image_size(image_data):
+        """
+        Returns the key for the largest image size available.
+        """
+        for size in ["l", "m", "s"]:
```

Comment:
We're also including `url_t` in the query params, should `t` (I'm assuming this means "tiny") be another option here as well?

Reply:
I'm not positive why we wouldn't, but we did not accept "t" in the original implementation and we also had a test that asserted that we get no `image_url` when only `url_t` is provided (using this resource: [image_data_no_image_url.json](https://github.com/WordPress/openverse-catalog/blob/main/tests/dags/providers/provider_api_scripts/resources/flickr/image_data_no_image_url.json)), so I'm at least cautious about changing it. Maybe in a separate issue?

Reply:
That works! And good find 🧐